pkgs/by-name/py/pyrefly/package.nix

```diff
@@ -0,0 +1,41 @@
+{
+  lib,
+  python3,
+  fetchPypi,
+  rustPlatform,
+  maturin,
+}:
+python3.pkgs.buildPythonApplication rec {
+  pname = "pyrefly";
+  version = "0.17.1";
+  format = "pyproject";
+
+  src = fetchPypi {
```

Comment:
We should probably add a comment explaining why we're pulling from the PyPI source (i.e. the source repo doesn't have a lockfile while the PyPI one does) since this is *unusual* for Nixpkgs

Reply:
Ah, sorry. I keep pushing while you're commenting... 😓 Is it bad cuz it stresses out the builders?

And yeah, I'll make the comment.

Reply:
It's fiiiiiine ;)

Reply:
@pluiedev Thanks. I commented something. Not sure how good it is: https://github.com/NixOS/nixpkgs/pull/412863/commits/37d02cd49909ce81f1c71cfd40360093a794865d